test/unit/user_serializer_test.rb

```diff
@@ -0,0 +1,15 @@
+require 'test_helper'
+
+module Shipit
+  class UserSerializerTest < ActiveSupport::TestCase
+    test 'includes anonymous key' do
+      serialized = UserSerializer.new(User.new).to_json
+      assert_equal false, JSON.parse(serialized)["anonymous"]
+    end
+
+    test 'sets anonymous to true for anonymous users' do
+      serialized = UserSerializer.new(AnonymousUser.new).to_json
+      assert_equal true, JSON.parse(serialized)["anonymous"]
```

Comment:
@byroot the test proves that it works, doesn't it?

Reply:
No, it proves that `UserSerializer` renders what you want, not that `UserSerializer` is used to serialize `AnonymousUser`.

AMS use a bunch of name based heuristic, so I assume that if I did add this empty `AnonymousUserSerializer` it was to comply to these.

If you want to prove it's useless (which is possible) then you need a controller test.

Reply:
Controller or webhook etc. By that I meant a higher level test that does go through the serializer selection logic.